src/server/controller/cycleData/persistNodeValue/validateNodeUpdates/validateNodeUpdates.ts

```diff
@@ -43,17 +45,46 @@ export const validateNodeUpdates = async (props: Props, client: BaseProtocol): P
       if (row.props.validateFns?.[cycle.uuid]) {
         // make sure in target table there's a matching column
         if (row.cols.find((c) => c.props.colName === colName)) {
-          // eslint-disable-next-line no-await-in-loop
-          const validation = await validateNode(
-            { assessment, cycle, countryIso, tableName, variableName, colName, row },
-            client
-          )
-          // eslint-disable-next-line no-await-in-loop
-          const node = await NodeRepository.updateValidation(
-            { assessment, cycle, tableName, variableName, countryIso, colName, validation },
-            client
-          )
-          value = node?.value
+          let isODPNode = false
+
+          // If we are handling tables 1a or 1b
+          if ([TableNames.extentOfForest, TableNames.forestCharacteristics].includes(tableName as TableNames)) {
+            // eslint-disable-next-line no-await-in-loop
+            const ODPYears = await OriginalDataPointRepository.getReservedYears(
+              { assessment, cycle, countryIso },
+              client
+            )
+
+            // We are on odp node if there is a year for current colName
+            isODPNode = ODPYears.map(String).includes(colName)
+          }
+
+          if (isODPNode) {
+            // Fetch the corresponding ODP value for the node update (used to update tables over websocket)
+            // eslint-disable-next-line no-await-in-loop
+            value = await getOriginalDataPointNode(
+              {
+                assessment,
+                cycle,
+                countryIso,
+                colName,
+                variableName,
+              },
+              client
+            )
+          } else {
+            // eslint-disable-next-line no-await-in-loop
+            const validation = await validateNode(
+              { assessment, cycle, countryIso, tableName, variableName, colName, row },
+              client
+            )
+            // eslint-disable-next-line no-await-in-loop
+            const node = await NodeRepository.updateValidation(
+              { assessment, cycle, tableName, variableName, countryIso, colName, validation },
+              client
+            )
+            value = node?.value
+          }
```

Comment:
can u explain what's happening here?

Reply:
We fetch the ODP value for the node if the criteria is met:
- table 1a/1b
- year for odp

Reply:
That I could see. But why ?
Can u explain why ?

Reply:
I think at the moment ODP validation happens only client side, isn't is so?
If yes, here we don't need to validate the ODP.

Reply:
so instead, in the parent where validateNodeUpdates is called we instead replace the found values? This one returns the non-odp values anyway, I must look deeper